nikola/nikola.py

```diff
@@ -1032,6 +1032,23 @@ def init_plugins(self, commands_only=False, load_all=False):
                 os.path.expanduser('~/.nikola/plugins'),
             ] + [utils.sys_encode(path) for path in extra_plugins_dirs if path]
 
+        compilers = defaultdict(set)
+        # Also add aliases for combinations with TRANSLATIONS_PATTERN
+        for compiler, exts in self.config['COMPILERS'].items():
+            for ext in exts:
+                compilers[compiler].add(ext)
+                for lang in self.config['TRANSLATIONS'].keys():
+                    candidate = utils.get_translation_candidate(self.config, "f" + ext, lang)
+                    compilers[compiler].add(candidate)
+
+        # Avoid redundant compilers
+        # Remove compilers that match nothing in POSTS/PAGES
```

Comment:
I removed `bad_compilers`, since the same information is already contained in `disabled_compilers` and nothing in core or `plugins` was using it.

Is the explanation in 53e5f6e good enough, or do you want something more detailed?